cmd/cstor-pool-mgmt/volumereplica/volumereplica.go

```diff
@@ -699,24 +700,127 @@ func GetAndUpdateSnapshotInfo(
 	}
 
 	// If CVR is in rebuilding go and get snapshot information
-	// from other replicas
-	//if cvr.Status.Phase == apis.CVRStatusRebuilding {
-	//	err = getAndUpdatePendingSnapshotsList(clientset, cvr)
-	//}
+	// from other replicas and add snapshots under pending snapshot list
+	if cvr.Status.Phase == apis.CVRStatusRebuilding ||
+		cvr.Status.Phase == apis.CVRStatusReconstructingNewReplica {
+		err = getAndAddPendingSnapshotList(clientset, cvr)
+		if err != nil {
+			return errors.Wrapf(err, "failed to update pending snapshots")
+		}
+	}
+
+	// It looks like hack but we must do this because of below reason
+	// - There might be chances in nth reconciliation CVR might be in Rebuilding
+	//   and pending snapshots added under CVR.Status.PendingSnapshots and after that
+	//   let us assume this pool is down meanwhile if snapshot deletion request
+	//   came and deleted snapshots in peer replicas. In next reconciliation if
+	//   CVR is Healthy then there might be chances that pending snapshots remains
+	//   as is to cover this corner case below check is required.
+	if cvr.Status.Phase == apis.CVRStatusOnline &&
```

Comment:
But `addOrDeleteSnapshotListInfo` will cover this right?

Reply:
`addOrDeleteSnapshotListInfo` will remove the `Pendingsnapshots`  only if exists on that replica.

Reply:
instead of this hack, call getAndAddPendingSnapshotList on every addEvent

Reply:
ok.. even though this looks hack, as per the 'Online' state definition, nothing is going to get rebuilt.. so, its fine setting to NULL..
how about adding a log with the list of snapnames that got removed?

Reply:
Added the log